shell/common/gin_converters/file_path_converter.h

```diff
@@ -23,9 +23,11 @@ struct Converter<base::FilePath> {
     if (val->IsNull())
       return true;
 
-    v8::String::Value str(isolate, val);
-    if (str.length() == 0) {
-      *out = base::FilePath();
+    // if the `val` is an empty string, set `out` to an empty FilePath
+    v8::Local<v8::Context> context = isolate->GetCurrentContext();
+    v8::Local<v8::String> str;
+    if (!val->ToString(context).ToLocal(&str) || str->Length() == 0) {
```

Comment:
I believe we only care about string inputs and not any object that is convertible to string, could this be

```suggestion
    if (!val->IsString())
      return true;
    // if the `val` is an empty string, set `out` to an empty FilePath
    v8::Local<v8::String> str = v8::Local<v8::String>::Cast(val);
    if (str->Length() == 0) {
```

Reply:
Even better! Less work for us.

Reply:
> I believe we only care about string inputs and not any object that is convertible to string

Seems like this is incorrect based on the failing checks.

Reply:
reverted in 51da82050ea6606d182c2cc4d3f8abf7e9a26680 but it's weird that we have tests passing in non-string values for as filepaths :confused:

Reply:
Haven't tested this yet, but my guess is that in 44bb8fa this paragraph returns true for any non-string type regardless of length, while current shipping code gets past this paragraph for any object where `v8::internal::Object::ConvertToString()` returns a nonempty string.

Maybe we can keep a modified 44bb8fa that returns false if it's not a string.

Reply:
force-pushed up 6a66b76a3ee27194e5b0c76c2c9d565dd1bcab95 to test that guess

Reply:
👍